### PR TITLE
atime, mtime and ctime setters

### PR DESCRIPTION
**Pull request by zeroos**

### CODE REVIEW

File: tests/test_atime_mtime_ctime.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+
+def stat_dict(path):
+    _, _, _, _, _, _, _, atime, _, ctime = stat(path)
+    return {"atime": atime, "ctime": ctime}
+
+
+def test_memory_atime_ctime():
+    # Collect information on what should be in the archive
+    tree = treestat('libarchive', stat_dict)
```
Comment (krzysztofzuraw) [inline tests/test_atime_mtime_ctime.py]: Can it be moved to fixture?

File: tests/test_atime_mtime_ctime.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    # Check the data
+    with memory_reader(buf) as archive2:
+        check_atime_ctime(archive2, tree)
+
+
+def test_file_atime_ctime(tmpdir):
+    archive_path = tmpdir.strpath+'/test.zip'
+
+    # Collect information on what should be in the archive
+    tree = treestat('libarchive', stat_dict)
```
Comment (krzysztofzuraw) [inline tests/test_atime_mtime_ctime.py]: Same here?

File: tests/test_atime_mtime_ctime.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+    # Read the archive and check that the data is correct
+    with file_reader(archive_path) as archive:
+        check_atime_ctime(archive, tree)
+
+
+def test_memory_time_setters():
+    # Create an archive of our libarchive/ directory
+
+    atimestamp = 1482144741.495628
```
Comment (kzuraw) [inline tests/test_atime_mtime_ctime.py]: can these numbers have more meaning? What does `1482144741.495628` represents? Seconds from 1970?

File: tests/test_atime_mtime_ctime.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    with memory_reader(buf2) as archive2:
+        for e in archive2:
+            assert e.atime == int(atimestamp)
+            assert e.mtime == int(mtimestamp)
+            assert e.ctime == int(ctimestamp)
+
+
+def test_file_time_setters(tmpdir):
+    # Create an archive of our libarchive/ directory
+    archive_path = tmpdir.strpath+'/test.zip'
```
Comment (kzuraw) [inline tests/test_atime_mtime_ctime.py]: Can it be made using tmpdir interface? http://doc.pytest.org/en/latest/tmpdir.html

File: tests/test_atime_mtime_ctime.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+    atimestamp = 1482144741.495628
+    mtimestamp = 1482155417.659017
+    ctimestamp = 1482145211.536858
+    with file_writer(archive_path, "zip") as archive1:
+        archive1.add_files('libarchive/')
+
+    with file_reader(archive_path) as archive1:
+        with file_writer(archive2_path, "zip") as archive2:
+            for e in archive1:
```
Comment (kzuraw) [inline tests/test_atime_mtime_ctime.py]: What is e?

File: libarchive/entry.py [hunk 6]
```diff
@@ -89,18 +91,37 @@ def issock(self):
     def isdev(self):
         return self.ischr or self.isblk or self.isfifo or self.issock
 
+    def _timestamp_sec_nsec(self, timestamp):
```
Comment (kzuraw) [inline libarchive/entry.py]: what does this function do?

File: tests/test_atime_mtime_ctime.py [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+    tree2 = copy(tree)
+    for e in archive:
+        epath = str(e).rstrip('/')
+        assert epath in tree2
+        estat = tree2.pop(epath)
+        assert e.atime == int(estat['atime'])
+        assert e.ctime == int(estat['ctime'])
+
+
+def stat_dict(path):
```
Comment (krzysztofzuraw) [inline tests/test_atime_mtime_ctime.py]: This can be made as a fixture

**Review by kzuraw — CHANGES_REQUESTED** [review]

**Review by kzuraw — APPROVED** [review]
Good job!

**Pull request merged**